### PR TITLE
test: add server tests and additional mocks

### LINKED ISSUE #160: Backfill `infrastructure/grpc/server.go` tests
There are none yet, but should be pretty straightforward as everything in there is mostly interface delegation

### PR DESCRIPTION
**Pull request by jmickey**
**What this PR does / why we need it**:

This PR backfills tests to the `infrastructure/grpc/server.go` file. Some additional error handling was added to the `server.go` functions being tested, as it was identified during the creation of the tests that this error handling was missing. Additional mocks have also been added.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #160 

**Checklist**:

- [x] squashed commits into logical changes
- [x] adds unit tests

### CODE REVIEW

**Comment by Callisto13** [comment]
Lovely tests 🎉 

While you are there, don't suppose you can remove the Note on line 21 of `infrastructure/grpc/server.go`? 😄

File: infrastructure/grpc/server_test.go [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+	grpcPkg "google.golang.org/grpc"
+
+	mvm1 "github.com/weaveworks/flintlock/api/services/microvm/v1alpha1"
+	"github.com/weaveworks/flintlock/api/types"
+	"github.com/weaveworks/flintlock/core/models"
+	"github.com/weaveworks/flintlock/infrastructure/grpc"
+	"github.com/weaveworks/flintlock/infrastructure/mock"
+)
+
+func TestServer_CreateMicroVM(t *testing.T) {
```
Comment (Callisto13) [inline infrastructure/grpc/server_test.go:19]: I don't think the validation call is covered here?

File: infrastructure/grpc/server_test.go [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+			tc.expect(cm.EXPECT(), qm.EXPECT())
+
+			ctx := context.Background()
+			svr := grpc.NewServer(cm, qm)
+			_, err := svr.DeleteMicroVM(ctx, tc.deleteReq)
+
+			if tc.expectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
```
Comment (Callisto13) [inline infrastructure/grpc/server_test.go:158]: This won't happen as there is no happy test case

File: infrastructure/grpc/server_test.go (hunk 1 above)
Comment (jmickey) [reply infrastructure/grpc/server_test.go:19]: It's not. I was unsure the best way to go about that - given the validation package itself is tested I didn't know if it was necessary to test the call to that package in this file?

File: infrastructure/grpc/server_test.go (hunk 2 above)
Comment (jmickey) [reply infrastructure/grpc/server_test.go:158]: I should add a happy path....

**Comment by jmickey** [comment]
@Callisto13 updated

**Review by Callisto13 — APPROVED** [review]
:shipit:

**Pull request merged**